### PR TITLE
Move ConcatDataset to common

### PR DESCRIPTION
**Pull request by aklife97**
Move `ConcatDataset` (earlier `ConcatTranslationDataset`) to common. 
The dataset was introduced in #2160

### CODE REVIEW

File: nemo/collections/common/data/dataset.py [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+            self.index_generator = ConcatDataset.temperature_generator
+            self.sampling_kwargs['temperature'] = sampling_temperature
+        elif sampling_technique == 'random':
+            self.index_generator = ConcatDataset.random_generator
+            self.sampling_kwargs['p'] = sampling_probabilities
+        elif sampling_technique == 'round-robin':
+            self.index_generator = ConcatDataset.round_robin_generator
+        else:
+            raise ValueError(f"Currently we only support sampling techniques in {supported_sampling_techniques}.")
+        self.N = 0
```
Comment (titu1994) [inline nemo/collections/common/data/dataset.py:70]: What is this variable?

File: nemo/collections/common/data/dataset.py [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+
+        for idx, dataset in enumerate(datasets):
+            isiterable = isinstance(dataset, IterableDataset)
+            if (isiterable and not self.kind == 'iterable') or (not isiterable and self.kind == 'iterable'):
+                raise ValueError("All datasets in ConcatDataset must be of the same kind (Iterable or Map).")
+
+            if self.kind == 'map':
+                self.N += len(dataset) // world_size
+            else:
+                self.N += len(dataset)
```
Comment (titu1994) [inline nemo/collections/common/data/dataset.py:85]: Can you give this variable a useful name rather than N

**Review by titu1994 — APPROVED** [review]
Looks fantastic, just could use a bit of refactoring of self.N

File: nemo/collections/common/data/dataset.py (hunk 2 above)
Comment (aklife97) [reply nemo/collections/common/data/dataset.py]: Done, changed it to length

**Pull request merged**